bp_be/src/include/bp_be_internal_if_defines.vh

```diff
@@ -0,0 +1,148 @@
+    logic[rv64_eaddr_width_gp-1:0]    br_tgt;                                                      \
+  }  bp_be_calc_result_s;                                                                          \
+
+/* Declare width macros so that clients can use structs in ports before struct declaration */
+`define bp_be_instr_metadata_width(branch_metadata_fwd_width_mp)                                   \
+  (bp_be_itag_width_gp                                                                             \
+   + rv64_eaddr_width_gp                                                                           \
+   + 1                                                                                             \
+   + $bits(bp_fe_exception_code_e)                                                                 \
+   + branch_metadata_fwd_width_mp                                                                  \
+   )                                                                                               \
+
+`define bp_be_issue_pkt_width(branch_metadata_fwd_width_mp)                                        \
+  (`bp_be_instr_metadata_width(branch_metadata_fwd_width_mp)                                       \
+   + rv64_instr_width_gp                                                                           \
+   + 4 * 1                                                                                         \
+   + 2 * rv64_reg_addr_width_gp                                                                    \
+   + rv64_reg_data_width_gp                                                                        \
+   )                                                                                               \
+
+`define bp_be_instr_operands_width                                                                 \
+  (3 * rv64_reg_data_width_gp)                                                                     \
+
+`define bp_be_pipe_stage_reg_width(branch_metadata_fwd_width_mp)                                   \
+  (`bp_be_instr_metadata_width(branch_metadata_fwd_width_mp)                                       \
+   + rv64_instr_width_gp                                                                           \
+   + `bp_be_instr_operands_width                                                                   \
+   + `bp_be_decode_width                                                                           \
+   )                                                                                               \
+
+`define bp_be_dep_status_width                                                                     \
+  (5 * 1 + rv64_reg_addr_width_gp)                                                                 \
+
+`define bp_be_calc_status_width(branch_metadata_fwd_width_mp)                                      \
+  (1                                                                                               \
```

Comment:
maybe add comments where each of these numbers come from, are they always going to be these values?

Reply:
Yeah it's just counting bits from the struct declaration.  I'll add a comment